fairseq/tasks/hubert_pretraining.py

```diff
@@ -107,19 +109,29 @@ class HubertPretrainingTask(FairseqTask):
     def __init__(
         self,
         cfg: HubertPretrainingConfig,
-        dictionaries: Dict[str, Dictionary],
     ) -> None:
         super().__init__(cfg)
 
         logger.info(f"current directory is {os.getcwd()}")
         logger.info(f"HubertPretrainingTask Config {cfg}")
-        self._dictionaries = dictionaries
+
+        self.fine_tuning = cfg.fine_tuning
+
+        if cfg.fine_tuning:
+            self.state.add_factory("ft_dictionaries", lambda: self.dictionaries_factory(cfg))
+            self._dictionaries = self.state.ft_dictionaries
+        else:
+            self.state.add_factory("pt_dictionaries", lambda: self.dictionaries_factory(cfg))
+            self._dictionaries = self.state.pt_dictionaries
+
+        if len(self._dictionaries) == 1:
+            self._target_dictionary = self._dictionaries[list(self._dictionaries)[0]]
+        else:
+            logger.info("Multiple Dictionaries, cannot pick single target.")
+            self._target_dictionary = {}
```

Comment:
@wnhsu How should multiple dictionaries be handled?

Reply:
This is how the `_target_dictionary` field was handled originally as well.

Reply:
the logic is a bit convolving when I first design this... so some basics:
1) fine-tuning uses only **one** dictionary
2) pre-training may use **multiple** dictionaries
3) fine-tuning builds the model with `task.target_dictionary`
4) pre-training builds the model with `task.dictionaries`
5) previously we have `task.target_dictionary = task.dictionaries[0]`


I'm going to propose that:
a) we change `HubertModel.build_model` to use `pt_dictionaries` instead of `dictionaries` for clarity
b) rename `dictionaries_factory` to `load_dictionaries`, and return `[Dictionary.load(f"{label_dir}/dict.{label}.txt") for label in cfg.labels]`. Note that we do not need to check if the file exists anymore since it was a workaround when we didn't save dictionaries into task.state before.
c) we do 
```
if cfg.fine_tuning:
  self.state_add_factory("target_dictionary", lambda: self.load_dictionaries(cfg)[0])
else:
  self.state_add_factory("dictionaries", lambda: self.load_dictionaries(cfg))
```
and remove L127-131. Note that now we use `@property` anymore for simplicity

Does that sound good?